### PR TITLE
Revert "Allocators should take a ubyte[] memory block instead of a void[] one"

### PR DESCRIPTION
**Pull request by aG0aep6G**
Reverts dlang/phobos#5322

### CODE REVIEW

**Comment by aG0aep6G** [comment]
Added a simple test for `Region` that shows why the reverted changes were bad. Other affected modules should probably get similar tests, but I'm not up for that work at the moment. It's disconcerting that those tests aren't there.

**Comment by aG0aep6G** [comment]
> Added a simple test

I've expanded the test to fail (reliably on my machine) when `void[]` is simply replaced with `ubyte[]`.

File: std/experimental/allocator/building_blocks/region.d [hunk 1]
```diff
@@ -357,6 +357,25 @@ struct Region(ParentAllocator = NullAllocator,
     // Destructor will free the memory
 }
 
+@system unittest
+{
+    import core.memory: GC;
```
Comment (JackStouffer) [inline std/experimental/allocator/building_blocks/region.d]: space before colon

Comment (aG0aep6G) [inline std/experimental/allocator/building_blocks/region.d]: fixed

**Comment by JackStouffer** [comment]
Seems like a previous PR was pulled that had failing style checks

**Comment by andralex** [comment]
We should probably keep https://github.com/dlang/phobos/pull/5322, refer to the discussion therein. Thx!

**Comment by MetaLang** [comment]
As @andralex said to keep https://github.com/dlang/phobos/pull/5322 I'm closing this.

**Pull request closed without merging**